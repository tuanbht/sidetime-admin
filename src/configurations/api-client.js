import axios from 'axios';

import { API_BASE_URL } from '../constants/api-paths';

const ApiClient = axios.create({
  baseURL: API_BASE_URL,
  headers: {
    Accept: 'application/json',
    'Content-Type': 'application/json',
    'Key-Inflection': 'camel',
  },
});

export default ApiClient;
