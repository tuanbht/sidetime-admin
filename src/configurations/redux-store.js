import axiosMiddleware from 'redux-axios-middleware';
import get from 'lodash/get';
import merge from 'lodash/merge';
import { configureStore } from '@reduxjs/toolkit';
import { persistReducer, persistStore, FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER } from 'redux-persist';
import storage from 'redux-persist/lib/storage';

import loadingApiActions from '../actions/loading-api-actions';
import reducers from '../reducers';
import authTokenActions from '../actions/auth-token-actions';

import AxiosClient from './api-client';

const persistConfig = {
  key: import.meta.env.VITE_WEBAPP_NAME,
  storage,
  whitelist: ['authToken'],
};

const persistedReducer = persistReducer(persistConfig, reducers);

const axiosMiddlewareConfig = {
  interceptors: {
    request: [
      ({ getState, dispatch }, request) => {
        const token = get(getState(), 'authToken');

        if (token) {
          merge(request, {
            headers: {
              Authorization: `Bearer ${token}`,
            },
          });
        }

        const selectedLanguage = get(getState(), 'lang');

        selectedLanguage &&
          merge(request, {
            headers: {
              Lang: selectedLanguage,
            },
          });

        dispatch(loadingApiActions.addLoadingApi());
        return request;
      },
    ],
    response: [
      {
        success: ({ dispatch }, response) => {
          dispatch(loadingApiActions.clearLoadedApi());
          return Promise.resolve(response);
        },
        error: ({ dispatch, getState }, response) => {
          const token = get(getState(), 'authToken');
          // Sign out if token is expired
          if (get(response, 'response.status') === 401 && token) {
            dispatch(authTokenActions.clearToken());
          }

          dispatch(loadingApiActions.clearLoadedApi());
          return Promise.reject(response);
        },
      },
    ],
  },
};

const middleware = (getDefaultMiddleware) =>
  getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
      ignoredActionPaths: ['payload.request'],
    },
  }).concat(axiosMiddleware(AxiosClient, axiosMiddlewareConfig));

const store = configureStore({
  reducer: persistedReducer,
  middleware,
  devTools: import.meta.env.DEV,
});

const persistor = persistStore(store);

export { store, persistor, middleware };
