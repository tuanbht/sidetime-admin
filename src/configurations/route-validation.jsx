import React from 'react';
import PropTypes from 'prop-types';

import ContainerLayout from '../components/container-layout';
import LoadingApiAnimation from '../components/loading-api-animation';

const RouteValidation = ({ container, ...props }) => (
  <React.Suspense fallback={<LoadingApiAnimation autoRun />}>
    <ContainerLayout container={container} {...props} />
  </React.Suspense>
);

RouteValidation.propTypes = {
  container: PropTypes.elementType.isRequired,
  privateRoute: PropTypes.bool,
  lockable: PropTypes.bool,
  hideDetectingAuthenticatedUser: PropTypes.bool,
  expertOnly: PropTypes.bool,
};

export default RouteValidation;
