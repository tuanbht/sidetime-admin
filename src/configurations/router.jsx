import React from 'react';
import { RouterProvider, createBrowserRouter, redirect } from 'react-router-dom';

import { NOT_FOUND_PATH, ROOT_PATH, SIGN_IN_PATH, SIGN_OUT_PATH, UNMATCHED_ROUTE_PATH } from '../constants/route-paths';
import SignIn from '../containers/sign-in';
import LoadingApiAnimation from '../components/loading-api-animation';
import ContainerLayout from '../components/container-layout';
import NotFoundPage from '../containers/not-found';

const router = createBrowserRouter([
  {
    id: 'root',
    path: ROOT_PATH,
    loader() {
      return { user: 1 };
    },
    element: <ContainerLayout container={SignIn} />,
    children: [
      {
        path: SIGN_IN_PATH,
        exact: true,
        element: <ContainerLayout container={SignIn} />,
        title: 'Sign in',
      },
    ],
  },
  {
    path: SIGN_OUT_PATH,
    loader() {
      return redirect(ROOT_PATH);
    },
  },
  {
    path: NOT_FOUND_PATH,
    element: <ContainerLayout container={NotFoundPage} />,
  },
  {
    path: UNMATCHED_ROUTE_PATH,
    loader() {
      return redirect(NOT_FOUND_PATH);
    },
  },
]);

const Router = () => <RouterProvider router={router} fallbackElement={<LoadingApiAnimation />} />;

export default Router;
