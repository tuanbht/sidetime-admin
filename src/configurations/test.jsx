import { configureStore } from '@reduxjs/toolkit';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { render } from '@testing-library/react';
import MockAdapter from 'axios-mock-adapter';
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter, Router } from 'react-router-dom';
import axiosMiddleware from 'redux-axios-middleware';

import reducer from '../reducers';

import axiosClient from './api-client';

const queryClient = new QueryClient({
  logger: {
    error: () => {},
  },
  defaultOptions: {
    queries: {
      retry: false,
      staleTime: 0,
    },
    mutations: {
      cacheTime: 0,
    },
  },
});

export const testStore = (initialState = {}) =>
  configureStore({
    preloadedState: initialState,
    reducer: reducer,
    middleware: [axiosMiddleware(axiosClient)],
  });

export const mockAxios = new MockAdapter(axiosClient);

export const renderApplication = (element, store = testStore(), entriesPath = '') =>
  render(
    <Provider store={store}>
      <QueryClientProvider client={queryClient}>
        <MemoryRouter initialEntries={[entriesPath]}>{element}</MemoryRouter>
      </QueryClientProvider>
    </Provider>,
  );

export const renderWithHistory = (element, customHistory, store = testStore()) =>
  renderApplication(<Router history={customHistory}>{element}</Router>, store);
