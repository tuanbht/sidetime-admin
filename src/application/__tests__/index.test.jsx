import { render, waitFor } from '@testing-library/react';
import React from 'react';
import { describe, expect, it } from 'vitest';

import Application from '..';
import { mockAxios } from '../../configurations/test';

describe('application', () => {
  it('renders template', async () => {
    mockAxios.onGet().reply(200, []);
    const { container } = render(<Application />);

    await waitFor(() => {
      expect(container).toMatchSnapshot();
    });
  });
});
