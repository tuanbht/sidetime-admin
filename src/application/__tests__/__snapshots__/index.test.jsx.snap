// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`application > renders template 1`] = `
<div>
  <div>
    <div>
      Intercom Provider
    </div>
    <div>
      <div
        class=""
        style="position: fixed; top: 0px; left: 0px; height: 2px; background: transparent; z-index: 99999999999; width: 100%;"
      >
        <div
          class=""
          style="height: 100%; background: rgb(241, 25, 70); transition: all 500ms ease; width: 100%;"
        >
          <div
            style="box-shadow: 0 0 10px #f11946, 0 0 10px #f11946; width: 5%; opacity: 1; position: absolute; height: 100%; transition: all 500ms ease; transform: rotate(3deg) translate(0px, -4px); left: 90%;"
          />
        </div>
      </div>
      <div
        class="Toastify"
      />
    </div>
    <div
      class=""
      style="position: fixed; top: 0px; left: 0px; height: 2px; background: transparent; z-index: 99999999999; width: 100%;"
    >
      <div
        class=""
        style="height: 100%; background: rgb(241, 25, 70); transition: all 500ms ease; width: 100%;"
      >
        <div
          style="box-shadow: 0 0 10px #f11946, 0 0 10px #f11946; width: 5%; opacity: 1; position: absolute; height: 100%; transition: all 500ms ease; transform: rotate(3deg) translate(0px, -4px); left: 90%;"
        />
      </div>
    </div>
  </div>
</div>
`;
